src/MongoDB.Driver.Core/Core/WireProtocol/CommandUsingCommandMessageWireProtocol.cs

```diff
@@ -262,54 +262,81 @@ private Type0CommandMessageSection<BsonDocument> CreateType0Section(ConnectionDe
         {
             var extraElements = new List<BsonElement>();
 
-            addIfNotAlreadyAdded("$db", _databaseNamespace.DatabaseName);
+            AddIfNotAlreadyAdded("$db", _databaseNamespace.DatabaseName);
 
             if (connectionDescription.IsMasterResult.ServerType != ServerType.Standalone
                 && _readPreference != null
                 && _readPreference != ReadPreference.Primary)
             {
                 var readPreferenceDocument = QueryHelper.CreateReadPreferenceDocument(_readPreference);
-                addIfNotAlreadyAdded("$readPreference", readPreferenceDocument);
+                AddIfNotAlreadyAdded("$readPreference", readPreferenceDocument);
             }
 
             if (_session.Id != null)
             {
-                addIfNotAlreadyAdded("lsid", _session.Id);
+                if (IsSessionAcknowledged())
+                {
+                    AddIfNotAlreadyAdded("lsid", _session.Id);
+                }
+                else
+                {
+                    if (_session.IsImplicit)
+                    {
+                        // do not set sessionId If session is implicit and write is unacknowledged
```

Comment:
nit: `If` does not need to be capitalized

Reply:
done